server/routerlicious/packages/services-client/src/rollingHash.ts

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISequencedDocumentMessage, ISequencedDocumentSystemMessage } from "@fluidframework/protocol-definitions";
+import { str } from "crc-32";
+
+export function getNextHash(message: ISequencedDocumentMessage, lastHash: string): string {
+    const messageWithData = message as ISequencedDocumentSystemMessage;
+
+    const clientIdProperty: keyof ISequencedDocumentSystemMessage = "clientId";
+    const sequenceNumberProperty: keyof ISequencedDocumentSystemMessage = "sequenceNumber";
+    const minimumSequenceNumberProperty: keyof ISequencedDocumentSystemMessage = "minimumSequenceNumber";
+    const clientSequenceNumberProperty: keyof ISequencedDocumentSystemMessage = "clientSequenceNumber";
+    const referenceSequenceNumberProperty: keyof ISequencedDocumentSystemMessage = "referenceSequenceNumber";
+    const typeProperty: keyof ISequencedDocumentSystemMessage = "type";
+    const timestampProperty: keyof ISequencedDocumentSystemMessage = "timestamp";
+    const dataProperty: keyof ISequencedDocumentSystemMessage = "data";
+
+    // Ordering of these fields decides the ordering of serialized message.
+    const fields: string[] = [
```

Comment:
All these fields and properties be moved outside this method so it won't create the object/strings every time.

Reply:
good point. Will do!